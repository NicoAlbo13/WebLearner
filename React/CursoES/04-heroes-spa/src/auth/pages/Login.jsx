import { useContext } from 'react'
import { useNavigate } from 'react-router-dom'
import { AuthContext } from '../context/AuthContext'

export const Login = () => {

  const {login} = useContext(AuthContext)
  const navigate = useNavigate()

  const handleLogin = ()=>{

    const path = localStorage.getItem('lastPath') || '/';

    login('Mr. Nooby')
    navigate(path, {replace:true})
  }

  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />

      <button 
      className="btn btn-primary"
      onClick={handleLogin}
      >
        Login
      </button>
    </div>
  )
}
