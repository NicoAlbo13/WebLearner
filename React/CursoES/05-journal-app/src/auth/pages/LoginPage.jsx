import { Link as RouterLink } from 'react-router'
import { Button, Grid, Link, TextField, Typography } from "@mui/material"
import { Google } from "@mui/icons-material";
import { AuthLayout } from '../layout/AuthLayout';

export const LoginPage = () => {
  return (
      <AuthLayout title='Login'>

        <form>
          <Grid container>
            <Grid size={12} sx={{mt: 2}}>
              <TextField 
              label='Mail' 
              type="email" 
              placeholder="mail@example.com"
              fullWidth
              />
            </Grid>

            <Grid size={12} sx={{mt: 2}}>
              <TextField 
              label='Password' 
              type="password" 
              placeholder="password"
              fullWidth
              />
            </Grid>

            <Grid container spacing={2} sx={{mb: 2, mt: 2}} size={12}>
              <Grid size={{xs: 12, sm: 6}}>
                <Button variant="contained" fullWidth>
                  Login
                </Button>
              </Grid>
              <Grid size={{xs: 12, sm: 6}}>
                <Button variant="contained" fullWidth>
                  <Google/>
                  <Typography sx={{ml: 1}}>Google</Typography>
                </Button>
              </Grid>
            </Grid>

            <Grid container direction='row' justifyContent='end' size={12}>
              <Link  component={RouterLink} color="inherit" to='/auth/register'>
                Create an account
              </Link>
            </Grid>

          </Grid>
        </form>

      </AuthLayout>
  )
}
