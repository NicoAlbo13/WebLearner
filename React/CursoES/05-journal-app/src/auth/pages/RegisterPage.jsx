import { Link as RouterLink } from 'react-router'
import { Button, Grid, Link, TextField, Typography } from "@mui/material"
import { AuthLayout } from '../layout/AuthLayout';

export const RegisterPage = () => {
  return (
      <AuthLayout title='Register'>

        <form>
          <Grid container>
            <Grid size={12} sx={{mt: 2}}>
              <TextField 
              label='Name' 
              type="text" 
              placeholder="Full Name"
              fullWidth
              />
            </Grid>

            <Grid size={12} sx={{mt: 2}}>
              <TextField 
              label='Mail' 
              type="email" 
              placeholder="mail@example.com"
              fullWidth
              />
            </Grid>

            <Grid size={12} sx={{mt: 2}}>
              <TextField 
              label='Password' 
              type="password" 
              placeholder="password"
              fullWidth
              />
            </Grid>

            <Grid container spacing={2} sx={{mb: 2, mt: 2}} size={12}>
              <Grid size={{xs: 12}}>
                <Button variant="contained" fullWidth>
                  Register
                </Button>
              </Grid>
            </Grid>

            <Grid container direction='row' justifyContent='end' size={12}>
              <Typography sx={{mr:1}}>Have an account?</Typography>
              <Link  component={RouterLink} color="inherit" to='/auth/login'>
                Login
              </Link>
            </Grid>

          </Grid>
        </form>

      </AuthLayout>
  )
}
