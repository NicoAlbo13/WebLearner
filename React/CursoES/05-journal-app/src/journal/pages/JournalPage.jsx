import { IconButton } from "@mui/material"
import { JournalLayout } from "../layout/JournalLayout"
import { EmptyView } from "../views/EmptyView"
import { NoteView } from "../views/NoteView"
import { AddOutlined } from "@mui/icons-material"

export const JournalPage = () => {
  return (
    <JournalLayout>
      <EmptyView/>
      {/* <NoteView/> */}

    <IconButton
      size="large"
      sx={{
        color: 'white',
        backgroundColor: 'error.main',
        ':hover': {backgroundColor: 'error.main', opacity: 0.8},
        position: 'fixed',
        right: 50,
        bottom: 50
      }}
    >
      <AddOutlined sx={{fontSize: 30}}/>
    </IconButton>

    </JournalLayout>
  )
}
