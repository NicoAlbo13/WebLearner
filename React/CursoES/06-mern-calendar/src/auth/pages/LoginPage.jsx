import { useEffect, useState } from 'react';
import Swal from 'sweetalert2';
import { useAuthStore } from '../../../hooks/useAuthStore';
import { useForm } from '../../../hooks/useForm';
import './login.css';

const loginForm = {
    loginEmail:'',
    loginPassword: ''
}

const registerForm = {
    registerName: '',
    registerEmail:'',
    registerPassword: '',
    registerPassword2: '',
}

export const LoginPage = () => {

    const { errorMessage, startLogin, startRegister } = useAuthStore();

    const [matchingPassword, setMatchingPassword] = useState('');
    const {loginEmail, loginPassword, onInputChange: onLoginChange} = useForm(loginForm);
    const {registerName, registerEmail, registerPassword, registerPassword2, onInputChange} = useForm(registerForm);

    const handleLogin = (e) => {
        e.preventDefault();

        startLogin({email: loginEmail, password: loginPassword});
    }

    const handleRegister = (e) => {
        e.preventDefault();
        if(registerPassword !== registerPassword2){
            Swal.fire('Register Error', 'Not matching passwords', 'error');
            return;
        }

        startRegister({name: registerName, email: registerEmail, password: registerPassword});
    }

    useEffect(() => {
        if(errorMessage!== undefined){
            Swal.fire('Error', errorMessage, 'error');
        }
    }, [errorMessage])

    useEffect(()=>{
        if(registerPassword !== registerPassword2 && registerPassword2.length > 0){
            setMatchingPassword('Password should match')
        }else{
            setMatchingPassword('');
        }
    }, [registerPassword2, registerPassword])

    return (
        <div className="container login-container">
            <div className="row">
                <div className="col-md-6 login-form-1">
                    <h3>Login</h3>
                    <form onSubmit={handleLogin}>
                        <div className="form-group mb-2">
                            <input
                                type="text"
                                className="form-control"
                                placeholder="Email"
                                name='loginEmail'
                                value={loginEmail}
                                onChange={onLoginChange}
                            />
                        </div>
                        <div className="form-group mb-2">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Password"
                                name='loginPassword'
                                value={loginPassword}
                                onChange={onLoginChange}
                            />
                        </div>
                        <div className="form-group mb-2">
                            <input
                                type="submit"
                                className="btnSubmit"
                                value="Login"
                            />
                        </div>
                    </form>
                </div>

                <div className="col-md-6 login-form-2">
                    <h3>Register</h3>
                    <form onSubmit={handleRegister}>
                        <div className="form-group mb-2">
                            <input
                                type="text"
                                className="form-control"
                                placeholder="Name"
                                name='registerName'
                                value={registerName}
                                onChange={onInputChange}
                            />
                        </div>
                        <div className="form-group mb-2">
                            <input
                                type="email"
                                className="form-control"
                                placeholder="Email"
                                name='registerEmail'
                                value={registerEmail}
                                onChange={onInputChange}
                            />
                        </div>
                        <div className="form-group mb-2">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Password"
                                name='registerPassword'
                                value={registerPassword}
                                onChange={onInputChange}
                            />
                        </div>

                        <div className="form-group mb-2">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Repeat Password"
                                name='registerPassword2'
                                value={registerPassword2}
                                onChange={onInputChange}
                            />
                        </div>

                        <div className="form-group mb-2">
                            <input
                                type="submit"
                                className="btnSubmit"
                                value="Create Account" />
                            <label style={{color: 'red'}}>{matchingPassword}</label>
                        </div>
                    </form>
                </div>
            </div>
        </div>
    )
}