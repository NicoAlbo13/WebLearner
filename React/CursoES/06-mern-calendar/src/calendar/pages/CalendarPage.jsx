import { useState } from 'react'
import { Calendar, dateFnsLocalizer } from 'react-big-calendar'
import 'react-big-calendar/lib/css/react-big-calendar.css'
import {format, addHours, parse, startOfWeek, getDay} from 'date-fns'
import enUS from 'date-fns/locale/en-US'

import { Navbar } from "../components/Navbar"
import { CalendarEvent } from '../components/CalendarEvent'
import { CalendarModal } from '../components/CalendarModal'
import { useUiStore } from '../../hooks/useUiStore'
import { useCalendarStore } from '../../hooks/useCalendarStore'
import { FabAddNew } from '../components/FabAddNew'
import { FabDelete } from '../components/FabDelete'

export const CalendarPage = () => {

  const { openDateModal } = useUiStore();
  const {events, setActiveEvent} = useCalendarStore();
  const [lastView, setLastView] = useState(localStorage.getItem('lastView') || 'month')

  const locales = {
    'en-US': enUS,
  }

  const localizer = dateFnsLocalizer({
    format,
    parse,
    startOfWeek,
    getDay,
    locales,
  })

  const eventStyleGetter = (event, start, end, isSelected)=> {
    const style = {
      backgroundColor: '#347CF7',
      borderRadius: '0px',
      opacity: 0.8,
      color: 'white'
    }

    return{
      style
    }
  }

  const onDoubleClick = (event) => {
    openDateModal()
  }

  const onSelect = (event) => {
    setActiveEvent(event);
  }

  const onViewChange = (event) => {
    localStorage.setItem('lastView', event);
    setLastView(event)
  }

  return (
    <>
      <Navbar/>
      <Calendar
        localizer={localizer}
        events={events}
        defaultView={lastView}
        startAccessor="start"
        endAccessor="end"
        style={{ height: '700px' }}
        eventPropGetter={eventStyleGetter}
        components={{
          event: CalendarEvent
        }}
        onDoubleClickEvent={onDoubleClick}
        onSelectEvent={onSelect}
        onView={onViewChange}
      />

      <CalendarModal/>
      <FabAddNew/>
      <FabDelete/>
    </>
  )
}
