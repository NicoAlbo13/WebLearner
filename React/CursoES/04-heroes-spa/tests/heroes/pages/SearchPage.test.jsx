import { fireEvent, render, screen } from "@testing-library/react";
import { SearchPage } from "../../../src/heroes/pages/SearchPage";
import { MemoryRouter, useNavigate } from "react-router-dom";

const mockedUseNavigate = jest.fn()

jest.mock("react-router-dom", ()=>({
    ...jest.requireActual("react-router-dom"),
    useNavigate: ()=> mockedUseNavigate,
}))

describe('test on SearchPage component', () => {

    beforeEach(()=>jest.clearAllMocks())

    test('should render with default correct values', () => {

        const {container} = render(
            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot();

    })

    test('should show Batman and input with the value of queryString', () => {

        const hero = 'batman'

        render(
            <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
                <SearchPage/>
            </MemoryRouter>
        )

        // screen.debug()

        const input = screen.getByRole('textbox');
        expect(input.value).toBe(hero)

        const img = screen.getByRole('img');
        expect(img.src).toContain(hero)

    })

    test('should show an error if hero not found (batman123)', () => {
        const hero = 'batman123'

        render(
            <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
                <SearchPage/>
            </MemoryRouter>
        )

        // screen.debug()
        expect(screen.getByText('Hero not found')).toBeTruthy()

    })

    test('should call navigate with searched hero onSubmit', () => {

        const mockedVal = 'wolverine'

        render(
            <MemoryRouter initialEntries={[`/search`]}>
                <SearchPage/>
            </MemoryRouter>
        )

        // screen.debug()
        const input = screen.getByRole('textbox')
        fireEvent.change(input, {target: {value: mockedVal}})
        const form = screen.getByLabelText('form')
        fireEvent.submit(form);

        expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${mockedVal}`)
    })

})
