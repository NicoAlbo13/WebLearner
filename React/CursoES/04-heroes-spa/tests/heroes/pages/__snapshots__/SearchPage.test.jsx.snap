// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on SearchPage component should render with default correct values 1`] = `
<div>
  <div
    class="animate__animated animate__fadeIn animate__fast"
  >
    <h1>
      Search
    </h1>
    <hr />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <h4>
          Searching
        </h4>
        <hr />
        <form
          aria-label="form"
        >
          <input
            autocomplete="off"
            class="form-control"
            name="searchText"
            placeholder="Search a hero"
            type="text"
            value=""
          />
          <button
            class="btn btn-outline-primary mt-3"
          >
            Send
          </button>
        </form>
      </div>
      <div
        class="col-7"
      >
        <h4>
          Results
        </h4>
        <hr />
        <div
          class="alert alert-primary animate__animated animate__zoomIn"
        >
          Search a hero
        </div>
      </div>
    </div>
  </div>
</div>
`;
